### PR TITLE
Integrate some_vehicle

### PR DESCRIPTION
**Pull request by janisozaur**
The name is :hankey: but I don't know any better.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
I remember this var vaguely. I think it can easily be named better if I have a read of it again

File: src/ride/vehicle.h [hunk 1]
```diff
@@ -385,6 +385,8 @@ const rct_vehicle_info *vehicle_get_move_info(int cd, int typeAndDirection, int
 uint16 vehicle_get_move_info_size(int cd, int typeAndDirection);
 bool vehicle_update_bumper_car_collision(rct_vehicle *vehicle, sint16 x, sint16 y, uint16 *spriteId);
 
+extern rct_vehicle *some_vehicle;
```
Comment (IntelOrca) [inline src/ride/vehicle.h]: Please use our convention for globals.... gCamalCase

**Comment by duncanspumpkin** [comment]
F64E04 is the current vehicle that is being updated. Really it should not be a global var it should be passed into functions.

File: src/ride/vehicle.c [hunk 2]
```diff
@@ -4033,7 +4034,7 @@ static void vehicle_update_doing_circus_show(rct_vehicle *vehicle)
 	if (RCT2_GLOBAL(0x00F64E34, uint8) == 0)
 		return;
 
-	totalTime = *(RCT2_ADDRESS(0x009A0AB4, uint16*)[vehicle->sub_state]);
+	totalTime = *(uint16*)((uintptr_t)(RCT2_ADDRESS(0x009A0AB4, uint32)[vehicle->sub_state]));
```
Comment (IntelOrca) [inline src/ride/vehicle.c]: I have fixed this in my local branch.

Comment (janisozaur) [inline src/ride/vehicle.c]: This wasn't supposed to be part of this commit, got here by mistake.

**Comment by IntelOrca** [comment]
call it `gCurrentVehicle`

**Comment by janisozaur** [comment]
rebased

**Pull request merged**